processing/src/main/java/io/druid/segment/virtual/BaseSingleValueDimensionSelector.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.virtual;
+
+import io.druid.segment.DimensionSelector;
+import io.druid.segment.data.IndexedInts;
+import io.druid.segment.data.ZeroIndexedInts;
+
+public abstract class BaseSingleValueDimensionSelector implements DimensionSelector
```

Comment:
It's Base because impls need to override `getValue()`. The idea is this selector always returns 0 as the int, but that int will return different strings via `lookupName` on different rows.

Cardinality being `CARDINALITY_UNKNOWN` is a signal that callers should expect this behavior (see javadocs for DimensionSelector.getValueCardinality).